File: decompiled/sources/freemarker/cache/TemplateLoader.java
package freemarker.cache;

import java.io.IOException;
import java.io.Reader;

public interface TemplateLoader {
    void closeTemplateSource(Object obj) throws IOException;

    Object findTemplateSource(String str) throws IOException;

    long getLastModified(Object obj);

    Reader getReader(Object obj, String str) throws IOException;
}

File: decompiled/sources/freemarker/cache/TemplateLoader.kt
package freemarker.cache

import java.io.IOException
import java.io.Reader

interface TemplateLoader {
    @Throws(IOException::class)
    fun closeTemplateSource(templateSource: Any)

    @Throws(IOException::class)
    fun findTemplateSource(name: String): Any?

    fun getLastModified(templateSource: Any): Long

    @Throws(IOException::class)
    fun getReader(templateSource: Any, encoding: String): Reader
}
